src/android/SimpleCameraPreview.java

```diff
@@ -128,51 +168,98 @@ public void onProviderDisabled(String provider) {
                 }
             };
 
-            if (cordova.hasPermission(Manifest.permission.ACCESS_FINE_LOCATION))
-                fetchLocation();
-            else
-                cordova.requestPermission(this, GEO_REQ_CODE, Manifest.permission.ACCESS_FINE_LOCATION);
+            fetchLocation();
             return true;
-        } catch(Exception e) {
+        } catch (Exception e) {
             e.printStackTrace();
             callbackContext.error(e.getMessage());
             return false;
         }
     }
 
-    private int getIntegerFromOptions(JSONObject options, String key){
+    private int getIntegerFromOptions(JSONObject options, String key) {
         try {
             return options.getInt(key);
         } catch (JSONException error) {
             return 0;
         }
     }
 
+    @Override
     public void onRequestPermissionResult(int requestCode, String[] permissions, int[] grantResults) {
-        if (grantResults.length < 1)
-            return;
-        if (grantResults[0] == PackageManager.PERMISSION_DENIED)
-            return;
+        if (requestCode == REQUEST_CODE_PERMISSIONS) {
+            boolean permissionsGranted = true;
+            if (grantResults.length > 0) {
+                for (int result : grantResults) {
+                    if (result != PackageManager.PERMISSION_GRANTED) {
+                        permissionsGranted = false;
+                        break;
+                    }
+                }
+            }
 
-        if (requestCode == GEO_REQ_CODE)
-            fetchLocation();
+            if (!permissionsGranted) {
+                boolean permissionAlwaysDenied = false;
+
+                for (String permission : permissions) {
+                    if (ActivityCompat.shouldShowRequestPermissionRationale(cordova.getActivity(), permission)) {
+                        cordova.requestPermissions(this, REQUEST_CODE_PERMISSIONS, REQUIRED_PERMISSIONS);
+                    } else {
+                        if (ActivityCompat.checkSelfPermission(cordova.getContext(), permission) == PackageManager.PERMISSION_DENIED) {
+                            permissionAlwaysDenied = true;
+                        }
+                    }
+                }
+
+                if (permissionAlwaysDenied) {
+                    AlertDialog.Builder builder = new AlertDialog.Builder(cordova.getContext());
+                    builder.setTitle("Permissions required")
+                            .setMessage("Go to app info > permissions and allow them")
```

Comment:
Not clear what to allow. Something along these lines :?
Please grant the Camera permission for this app from your Settings. ?

Reply:
Sure the camera permission